ivy/array/utility.py

```diff
@@ -12,17 +12,17 @@ class ArrayWithUtility(abc.ABC):
     def all(
         self: ivy.Array,
         axis: Optional[Union[int, Tuple[int], List[int]]] = None,
-        keepdims: bool = False,
+        keepdims: Optional[bool] = False,
```

Comment:
There might be some conflicting information in some areas in the deep dive section, and I'm sorry if that's the case, but actually I have recently learnt that `Optional` should only be used to flag that `None` is a permitted value for the argument, as explained in the answer by Tomasz Bartkowiak [here](https://stackoverflow.com/questions/38727520/how-do-i-add-default-parameters-to-functions-when-using-type-hinting). Therefore, we should only use `Optional` for arguments which have a default value of `None`.